src/network/taraxa_capability.cpp

```diff
@@ -1155,11 +1155,21 @@ void TaraxaCapability::sendTransactions(NodeID const &_id, std::vector<taraxa::b
   LOG(log_nf_trx_prp_) << "sendTransactions " << transactions.size() << " to " << _id;
   RLPStream s(transactions.size());
   taraxa::bytes trx_bytes;
+  uint32_t transaction_counter = 0;
   for (auto transaction : transactions) {
     trx_bytes.insert(trx_bytes.end(), std::begin(transaction), std::end(transaction));
+    transaction_counter++;
+    if (transaction_counter == MAX_TRANSACTIONS_IN_PACKET) {
+      s.appendRaw(trx_bytes, transaction_counter);
+      transaction_counter = 0;
+      sealAndSend(_id, TransactionPacket, move(s));
+      trx_bytes.clear();
+    }
+  }
+  if (transaction_counter) {
```

Comment:
Just a note: I think with do while(transaction_counter <= transactions.size()) you could get rid of this whole condition and two sealAndSend calls

Reply:
I changed the implementation to actually send max 1000 transactions each interval (current this is 100ms) because only splitting the packets still basically sends all transactions at once